internal/controller/managedcluster_controller.go

```diff
@@ -683,6 +695,357 @@ func (r *ManagedClusterReconciler) machinesAvailable(ctx context.Context, namesp
 	return len(itemsList.Items) != 0, nil
 }
 
+func (r *ManagedClusterReconciler) reconcileCredentialPropagation(ctx context.Context, l logr.Logger, managedCluster *hmc.ManagedCluster) (ctrl.Result, error) {
+	l.Info("Reconciling CCM credentials propagation")
+	var err error
+	defer func() {
+		err = errors.Join(err, r.updateStatus(ctx, managedCluster))
+	}()
+
+	providers, err := r.getInfraProviders(ctx, managedCluster.Namespace, managedCluster.Spec.Template)
+	if err != nil {
+		managedCluster.Status.CCMSecretReconciled = false
```

Comment:
you can initially set it to False in the beginning of the reconcile and only change it to True once it succeed

Reply:
i think it is not required at all regarding [this comment](https://github.com/Mirantis/hmc/pull/405/files#r1801166440), having detailed conditions along in the status